### PR TITLE
Update contributor docs & PR template

### PR DESCRIPTION
**Pull request by jeff-mccoy**
Update PR template and external contributor rules for PRs. Cleans up some wording on the contributor guide as well.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging 

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,27 +1,18 @@
...
+...
 
 ## Related Issue
 
-<!--- This project prefers to accept pull requests related to open issues -->
-<!--- If suggesting a new feature or change, please discuss it in an issue first -->
-<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
-
-Fixes # (issue)
+...
```
Comment (Racer159) [inline .github/pull_request_template.md:7]: I like the suggestion to put a "fixes issue" here and don't think we should drop that.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,27 +1,18 @@
...
 ## Type of change
 
-<!-- Please delete options that are not relevant -->
-
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] New feature (non-breaking change which adds functionality)
-- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-
-## Checklist before merging
+- [ ] Other (security config, docs update, etc)
```
Comment (Racer159) [inline .github/pull_request_template.md:14]: Random extra spaces?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -2,71 +2,58 @@
...
+
+This section dives deeper into how we test Zarf
 
-1. Install [pre-commit](https://pre-commit.com/)
-1. Install [go](https://golang.org/)
-1. Run `pre-commit install` in the repo to install the pre-commit hooks. This will make the hooks run automatically each time you `git commit`. If you want to skip the hooks for any reason you can run `git commit --no-verify` to skip them.
+### (Optional) Pre-Commit Hooks and Linting
 
-> ℹ️ **HINT:** *Consider [automatically enabling the hooks in every Git repository](https://pre-commit.com/#automatically-enabling-pre-commit-on-repositories)*
+In this repo you can optinally use [pre-commit](https://pre-commit.com/) hooks for automated validation and linting, but if not the CI will run these checks for you.
```
Comment (Racer159) [inline CONTRIBUTING.md:44]:
```suggestion
In this repo you can optinally use [pre-commit](https://pre-commit.com/) hooks for automated validation and linting, but if not CI will run these checks for you.
```

File: .github/pull_request_template.md (hunk 1 above)
Comment (Racer159) [reply .github/pull_request_template.md:7]: People will often skim the contributor guidelines but the PR template is something that will be shoved in their face and I'd rather encourage best practices like this more directly.

**Review by Racer159 — CHANGES_REQUESTED** [review]

File: .github/pull_request_template.md (hunk 2 above)
Comment (jeff-mccoy) [reply .github/pull_request_template.md:14]: which space?

Comment (Racer159) [reply .github/pull_request_template.md:14]: NVM, I was holding it wrong.

**Review by Racer159 — APPROVED** [review]

**Review by YrrepNoj — APPROVED** [review]

**Pull request merged**